mods/player_api/api.lua

```diff
@@ -127,12 +122,30 @@ function player_api.set_animation(player, anim_name, speed)
 	if player_data.animation == anim_name and player_data.animation_speed == speed then
 		return
 	end
-	local previous_anim_equals = (model.animations[player_data.animation] or {})._equals
+	local previous_anim = model.animations[player_data.animation] or {}
 	local anim = model.animations[anim_name]
 	player_data.animation = anim_name
 	player_data.animation_speed = speed
+	-- Set the animation seen by everyone else
 	player:set_animation(anim, speed, animation_blend)
-	if anim._equals ~= previous_anim_equals then
+	-- If necessary change the local animation (only seen by the client of *that* player)
+	-- `override_local` <=> suspend local animations while this one is active
+	if anim.override_local ~= previous_anim.override_local then
+		if anim.override_local then
+			player:set_local_animation(
+				anim, anim, anim, anim,
```

Comment:
Isn't setting the same animation for standing, walking, mining and walkmine a dirty hack?